src/ModelContextProtocol/Server/AIFunctionMcpServerTool.cs

```diff
@@ -195,57 +195,59 @@ public override async Task<CallToolResponse> InvokeAsync(
             };
         }
 
-        switch (result)
+        return result switch
         {
-            case null:
-                return new()
-                {
-                    Content = []
-                };
-
-            case string text:
-                return new()
-                {
-                    Content = [new() { Text = text, Type = "text" }]
-                };
-
-            case TextContent textContent:
-                return new()
-                {
-                    Content = [new() { Text = textContent.Text, Type = "text" }]
-                };
-
-            case DataContent dataContent:
-                return new()
-                {
-                    Content = [new()
+            null => new()
+            {
+                Content = []
+            },
+            string text => new()
+            {
+                Content = [new() { Text = text, Type = "text" }]
+            },
+            TextContent textContent => new()
+            {
+                Content = [new() { Text = textContent.Text, Type = "text" }]
+            },
+            DataContent dataContent => new()
+            {
+                Content = [new()
                     {
                         Data = dataContent.GetBase64Data(),
                         MimeType = dataContent.MediaType,
                         Type = dataContent.HasTopLevelMediaType("image") ? "image" : "resource",
                     }]
-                };
+            },
+            string[] texts => new()
+            {
+                Content = [.. texts.Select(x => new Content() { Type = "text", Text = x ?? string.Empty })]
+            },
 
-            case string[] texts:
-                return new()
-                {
-                    Content = texts
-                        .Select(x => new Content() { Type = "text", Text = x ?? string.Empty })
-                        .ToList()
-                };
+            IEnumerable<AIContent> contentItems => new()
+            {
+                Content = [.. contentItems.Select(static item => item switch
+                        {
+                            TextContent textContent => new Content() { Type = "text", Text = textContent.Text },
+                            DataContent dataContent => new Content()
+                            {
+                                Data = dataContent.GetBase64Data(),
+                                MimeType = dataContent.MediaType,
+                                Type = dataContent.HasTopLevelMediaType("image") ? "image" : "resource",
+                            },
+                            _ => new Content() { Type = "text", Text = item.ToString() ?? string.Empty }
```

Comment:
ToString on an AIContent other than a TextContent is not going to be meaningful.

This should probably fall back to JsonSerializer.Serialize on the item just like the default case does below for a single item.